[wip] правила принятия решений (issue/PR)

glitchcore opened the pull request:
Сделайте с этим что-нибудь

glitchcore committed:
правила принятия решений

kaero commented:
/a

sntxerror committed:
Правит правила принятия issue и правила именования тегов

sntxerror commented:
посмотрите новые правки

/d

glitchcore commented on README.md:
```diff
@@ -6,15 +6,28 @@
-Правила ведения дискуссий, принятия PR находятся в разработке и обсуждаются в [issue #2](https://github.com/b4ck5p4c3/wiki/issues/2)
+### Правила принятия PR/Issues
+
+Каждый PR/Isuue должен быть обоснованым, опираться на принципы и правила b4cksp4ce, на факты и логику.
```
Я бы убрал про факты и логику — у нас будет отдельная история про правила ведения дискуссий. Тем более что некоторые issue из разряда «хочу устроить завтра просмотр кино» довольно сложно логически обосновать

glitchcore commented on README.md:
```diff
@@ -6,15 +6,28 @@
+
+При одобрении, участники могут принять Isuue/PR командой `/a` (accept).
+
+Таймер перезапускается токльо при смене флага d -> a или a -> d. 
```
Еще таймер перезапускается при изменении содержимого PR или описания задачи.

Кстати, описание задачи может менять только автор?

kaero replied:
> описание задачи может менять только автор?

нет, но история изменений комментов и описания ведется

glitchcore replied:
Мб нужно добавить правило о том, что описание может менять только автор?

glitchcore commented on README.md:
```diff
@@ -6,15 +6,28 @@
+
+Таймер перезапускается токльо при смене флага d -> a или a -> d. 
+
+Если есть внешние факторы влияющие на сроки выполнения/принятия, таймер может быть переопределен указанием сроков PR/Isuue командой `/deadline YYYY-MM-DDThh:mm` с указанием времени в формате [ISO 8601](https://wikipedia.org/wiki/ISO_8601).
```
Должна быть возможность отменить deadline

kaero replied:
/deadline 2600 установит дедлайн на 2600 год, что фактически эквивалентно его отмене, не плодите сущности

glitchcore replied:
Ок

glitchcore commented on README.md:
```diff
@@ -6,15 +6,28 @@
+
+Каждый PR/Isuue должен быть обоснованым, опираться на принципы и правила b4cksp4ce, на факты и логику.
+
+Любому PR/Issue при открытии устанавливается суточный таймер на автоматическое принятие/закрытие.
```
Не любому, а тому который содержит discuss

glitchcore commented:
Еще: я бы здесь добавил про то, что рекомендуется issue создавать сразу с оформленным предложением, чтобы принятие issue означало принятие предложения. Можно создать с вопросом, но тогда наверное в процессе обсуждения поменять описание так, чтобы в его тексте появился ответ, который можно будет принять или отклонить

glitchcore commented:
Ну и наверное важно отметить, что правила принятия PR, сформированые в этом PR принимаются и к этому PR *утонул в сингулярности*

Commit:
изменения в результате обсуждения #30

- ввел термин "команда", так как он использовался, но не был определен
- выделил термины курсивом
- определил термин "таймер принятия решения"
- удалил упоминание фактов и логики https://github.com/b4ck5p4c3/wiki/pull/30/files#r341053657
- добавил упоминание тэга discuss https://github.com/b4ck5p4c3/wiki/pull/30/files#r341054324
- всячески уточнил формулировки, поправил опечатки, форматирование

Commit:
Удалил неловкий пробел

kaero commented:
Замечания выше устранены, кроме предложения ограничения на изменение описания только автором https://github.com/b4ck5p4c3/wiki/pull/30#discussion_r341394891

Я не считаю, что оно требуется, так как инициатор дискуссии может забросить ее, но другой заинтересованный может продолжать.

Commit:
избавился от бессмысленного заголовка

Commit:
добавил абзац про формулирование issue 

и поддержание формулировки в актуальном состоянии

kaero commented:
Добавил абзац по замечанию https://github.com/b4ck5p4c3/wiki/pull/30#issuecomment-548302110

kaero commented:
/a

glitchcore commented on README.md:
```diff
@@ -4,15 +4,40 @@
+
+### Правила принятия *PR* и issues
+
+Каждый *PR* или issue, несогласие с предлагаемыми изменениями, предложением или комментарием должны быть обоснованым, опираться на принципы и правила b4cksp4ce.
```
Предлагаю изменение 

`Текст каждого *PR* или issue, а также комментарии в их обсуждении должны быть обосноваными, опираться на принципы и правила b4cksp4ce.`
/d

glitchcore commented on README.md:
```diff
@@ -4,15 +4,40 @@
+
+Каждый *PR* или issue, несогласие с предлагаемыми изменениями, предложением или комментарием должны быть обоснованым, опираться на принципы и правила b4cksp4ce.
+
+Для *PR* или issue с меткой `discuss` при открытии устанавливается *таймер принятия решения* на 24 часа. 
```
Я посмотрел на текущий опыт прохождения обсуждений и у меня появилось предложение, подкупающее своей новизной: не запускать таймер по тегу, а делать это явно.

Если участник обсуждения считает, что консенсус о разрешении/отклонении вопроса принят, то командой предлагает его закрытие по прошествии заданного времени.
`/a` — принять предложение
`/d` — отклонить предложение.

Если дискуссия продолжается, можно отменить таймер командой `/stop`

kaero replied:
Тэг действительно можно отменить, и запускать по первой команде.

`/stop` предлагаю обсудить и внедрить или нет отдельным issue.

glitchcore commented:
В целом все это описание команды выглядит для меня довольно громоздко и по-канцелярски.

Мне кажется, надо написать по-человечески:
1. Обсуждение вопроса считается завершенным, если участники договорились об этом.
2. Текст issue/PR и комментарии должны быть обосноваными, опираться на принципы и правила b4cksp4ce.
3. Когда обсуждение вопроса считается завершенным, issue/PR предлагается к закрытию/отклонению: устанавливается интервал на 24 часа. Если за это время ситуация не меняется, issue/PR закрывается/отклоняется.
4. Вопросы требующие срочного решения (в силу внешних факторов) закрываются в установленное время.

(а дальше уже расписать про команды)

kaero commented:
> В целом все это описание команды выглядит для меня довольно громоздко и по-канцелярски.

Согласен, что получилось громоздко, но предыдущая ревизия не была полна и понятна для человека, который не был участником устного обсуждения.

Попробую переформулировать ещё раз.

glitchcore committed:
refactor text

glitchcore commented:
Перелопатил текст, посмотрите, добавил человеческое описание в начало. Кажется, мы уже со всех сторон обсосали эту тему, если текст не совсем ужасный, то предлагаю принимать
/a

z4fd commented on README.md:
```diff
@@ -6,38 +6,43 @@
 Текст данной wiki, в том числе правила и принципы, могут изменяться с помощью pull request (далее *PR*). Обсуждение изменений происходит в *PR*.
 
-### Команды в *PR* и issues
+Текст каждого *PR* или issue, а также комментарии в их обсуждении должны быть обосноваными, опираться на принципы и правила b4cksp4ce.
```
Примерный вариант, еще попробую переписать:

Текст этой wiki, так же как наши правила и принципы, меняются с помощью pull request (далее PR). Изменения обсуждаем в комментариях к PR.

PR, issue и комментарии в обсуждении должны быть аргументированными и опираться на принципы и правила b4cksp4ce.

Если issue решает проблему, то рекомендуем добавить описание проблемы и предложение как решить. Если предложения не будет, то решение появиться при обсуждении. То что было решено при обсуждении, перед принятием PR добавляется в описание issue.

z4fd commented:
Оh, shi ~ надо еще очеловечивать, все еще много канцеляризма, кинул в ревью первый вариант. Попробую еще отрефакторить 

/d

glitchcore commented:
> Оh, shi ~ надо еще очеловечивать, все еще много канцеляризма, кинул в ревью первый вариант. Попробую еще отрефакторить
> 
> /d

В целом мне нравится, но сейчас выкачу как есть, потому что правила уже фактически действуют и мне хочется видеть их в вики. @z4fd закинь эти правки отдельным PR?

z4fd commented:
> > Оh, shi ~ надо еще очеловечивать, все еще много канцеляризма, кинул в ревью первый вариант. Попробую еще отрефакторить
> > /d
> 
> В целом мне нравится, но сейчас выкачу как есть, потому что правила уже фактически действуют и мне хочется видеть их в вики. @z4fd закинь эти правки отдельным PR?

Оке, еще подумаю и закину отдельным пулом
